src/Foundation/NSUrl.cs

```diff
@@ -30,6 +30,32 @@ public NSUrl (string path, string relativeToUrl)
 		{
 		}
 
+		public NSUrl (string urlString)
+		{
+			Handle = _Init (urlString);
+			if (Handle == IntPtr.Zero)
+				throw new ArgumentException ("Malformed url");
+		}
+
+		public NSUrl (string urlString, NSUrl relativeToUrl)
+		{
+			Handle = _Init (urlString, relativeToUrl);
+			if (Handle == IntPtr.Zero)
+				throw new ArgumentException ("Malformed url");
+		}
+
+		// Call these instead of obj-c static versions since these throw
+		// if nil returned due to malformed input
```

Comment:
See - https://bugzilla.xamarin.com/show_bug.cgi?id=40861

Could not initialize an instance of the type 'MonoMac.Foundation.NSUrl': the native 'initWithString:' method returned nil.\nIt is possible to ignore this condition by setting MonoTouch.ObjCRuntime.Class.ThrowOnInitFailure to false."